src/map/entities/mobentity.cpp

```diff
@@ -518,9 +518,9 @@ void CMobEntity::Spawn()
     uint8 level = m_minLevel;
 
     // Generate a random level between min and max level
-    if (m_maxLevel != m_minLevel)
+    if (m_maxLevel > m_minLevel)
     {
-        level += dsprand::GetRandomNumber(0, m_maxLevel - m_minLevel);
+        level += dsprand::GetRandomNumber(0, m_maxLevel - m_minLevel + 1);
```

Comment:
Like KJ said, we can make this just roll a range between min/max level. It's a templated function so you probably just need to declare the type explicitly for uint8.

Like this:
`level = dsprand::GetRandomNumber<uint8>(m_minLevel, m_maxLevel + 1);`

Reply:
didn't work.  it works if i cast both inputs from uint8 to int.
```uint8 level = dsprand::GetRandomNumber((int)m_minLevel, (int)m_maxLevel + 1);```

is that ok?